mne_bids/path.py

```diff
@@ -941,3 +941,99 @@ def _path_to_str(var):
                          f"pathlib.Path objects. Found type {type(var)}.")
     else:
         return str(var)
+
+
+def _filter_fnames(fnames, *, subject=None, session=None, task=None,
+                   acquisition=None, run=None, processing=None, recording=None,
+                   space=None, split=None, kind=None, extension=None):
+    """Filter a list of BIDS filenames based on BIDS entity values."""
+    sub_str = f'sub-{subject}' if subject else r'sub-([^_]+)'
+    ses_str = f'_ses-{session}' if session else r'(|_ses-([^_]+))'
+    task_str = f'_task-{task}' if task else r'(|_task-([^_]+))'
+    acq_str = f'_acq-{acquisition}' if acquisition else r'(|_acq-([^_]+))'
+    run_str = f'_run-{run}' if run else r'(|_run-([^_]+))'
+    proc_str = f'_proc-{processing}' if processing else r'(|_proc-([^_]+))'
+    rec_str = f'_rec-{recording}' if recording else r'(|_rec-([^_]+))'
+    space_str = f'_space-{space}' if space else r'(|_space-([^_]+))'
+    split_str = f'_split-{split}' if split else r'(|_split-([^_]+))'
+    # XXX do we want MRI here?
+    kind_str = f'_{kind}' if kind else r'_(meg|eeg|ieeg|mri)'
```

Comment:
@hoechenberger shall we use ALLOWED_MODALITY_EXTENSIONS to build this string here?

Reply:
Yep, did that now

Reply:
Errm, using ALLOWED_FILENAME_KINDS, I mean.